src/parser.lalrpop

```diff
@@ -121,6 +121,11 @@ Ann: Annotation = {
 	"@" <s:NameDecl> => Annotation::new(s),
 }
 
+TypeName: CascadeString = {
+	Symbol,
+	<start: @L> <l: TypeName> "." <r: Symbol> <end: @R> => CascadeString::new([l.as_ref(), ".", r.as_ref()].concat(), start..end)
```

Comment:
Why `Symbol` but not `NameDecl` instead? It would be nice to forbid the use of `BuiltInType` names and other keywords (e.g. "let", "module", "if"…).

Reply:
I don't think we want to forbid built in types everywhere that TypeName is used.  For example, the definition of List or Argument both refer to TypeName, but I think built in types can work fine in those contexts.

There are probably a subset of cases here where banning built-in types makes sense (ie a TypeName such as `foo.domain`), but I think that that will definitely fail in the actual compilation because it will never exist because it's impossible to declare a resource named `domain` (or if not, it should be and that's the bug to fix).  So maybe we should enforce this more strictly at the parser, but that might be low priority follow-up work, I think.

I think keywords that exist as strings in parser.lalrpop match as strings first in the lexer and therefore can't match Symbol.